pkg/jobs/registry.go

```diff
@@ -530,15 +530,14 @@ func (r *Registry) CreateStartableJobWithTxn(
 	var span *tracing.Span
 	var execDone chan struct{}
 	if !alreadyInitialized {
-		// Construct a context which contains a tracing span that follows from the
-		// span in the parent context. We don't directly use the parent span because
-		// we want independent lifetimes and cancellation. For the same reason, we
-		// don't use the Context returned by ForkSpan.
+		// Construct a context which contains a root tracing span. This allows
+		// tracking all the spans associated with job execution using the trace_id
+		// of the newly constructed root span.
+		// Using a new context and span allow for independent lifetimes and
+		// cancellation.
 		resumerCtx, cancel = r.makeCtx()
-		_, span = tracing.ForkSpan(ctx, "job")
-		if span != nil {
-			resumerCtx = tracing.ContextWithSpan(resumerCtx, span)
-		}
+		resumerCtx, span = r.settings.Tracer.StartSpanCtx(resumerCtx, "job",
+			tracing.WithForceRealSpan())
```

Comment:
Along this line, one idea is to give jobs a verbosity level of their own that they can declare when creating their job record. That way particular types of jobs can opt-in to be traced (e.g. we're pretty confident there won't be a crazy amount of backup/restore jobs so those jobs can always enable tracing).

Reply:
This is a good idea, I haven't added it in yet because I want to flesh this out a little.

A span can either be a noop (this is the default unless tracing is on) or a "real" span. A noop span records nothing. A "real" span can further be in verbose mode or not. A non-verbose, real span allows for recording structured payloads but discards all free form logging. A verbose, real span allows for all forms of recording. In the initial google doc, knz had suggested we stick to structured payloads for tracing job execution.

So I'm inclined to allow jobs to opt-in to being a real, root span, but not allowing them to choose between verbose/non-verbose. What do you think?

Reply:
That sounds good, my mention of jobs having a "verbosity" was to air on the side of making that flag an enum in the proto rather than a bool so that we can adjust later on if we want to add more fine-grained control.